javax-money/src/main/java/com/fasterxml/jackson/datatype/money/JavaxMoneyModule.java

```diff
@@ -0,0 +1,115 @@
+package com.fasterxml.jackson.datatype.money;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleDeserializers;
+import com.fasterxml.jackson.databind.module.SimpleSerializers;
+import org.apiguardian.api.API;
+
+import javax.money.CurrencyUnit;
+import javax.money.MonetaryAmount;
+import javax.money.format.MonetaryFormats;
+import java.util.Objects;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.apiguardian.api.API.Status.STABLE;
+
+
+@API(status = STABLE)
+public final class JavaxMoneyModule extends Module {
+
+    private final AmountWriter<?> writer;
+    private final FieldNames names;
+    private final MonetaryAmountFormatFactory formatFactory;
+    private final MonetaryAmountFactory<? extends MonetaryAmount> amountFactory;
+    private final Class implementationClass;
+
+    public JavaxMoneyModule() {
+        this(new DecimalAmountWriter(), FieldNames.defaults(), MonetaryAmountFormatFactory.NONE, null, null);
+    }
+
+    private <T extends MonetaryAmount> JavaxMoneyModule(final AmountWriter<?> writer,
+                                                        final FieldNames names,
+                                                        final MonetaryAmountFormatFactory formatFactory, Class<T> implementationClass,
+                                                        final MonetaryAmountFactory<T> amountFactory) {
+
+        this.writer = writer;
+        this.names = names;
+        this.formatFactory = formatFactory;
+        this.implementationClass = implementationClass;
+        this.amountFactory = amountFactory;
+    }
+
+    @Override
+    public String getModuleName() {
+        return JavaxMoneyModule.class.getSimpleName();
+    }
+
+    @Override
+    public Version version() {
+        return PackageVersion.VERSION;
+    }
+
+    @Override
+    public void setupModule(final SetupContext context) {
+        final SimpleSerializers serializers = new SimpleSerializers();
+        serializers.addSerializer(CurrencyUnit.class, new CurrencyUnitSerializer());
+        serializers.addSerializer(MonetaryAmount.class, new MonetaryAmountSerializer(names, writer, formatFactory));
+        context.addSerializers(serializers);
+
+        final SimpleDeserializers deserializers = new SimpleDeserializers();
+        deserializers.addDeserializer(CurrencyUnit.class, new CurrencyUnitDeserializer());
+        if (Objects.nonNull(this.amountFactory) && Objects.nonNull(this.implementationClass)) {
+            deserializers.addDeserializer(MonetaryAmount.class, new MonetaryAmountDeserializer<>(amountFactory, names));
+            // for reading into concrete implementation types
+            deserializers.addDeserializer(implementationClass, new MonetaryAmountDeserializer<>(amountFactory, names));
+        }
```

Comment:
I believe it could be replaced with the canonical way to build a new MonetaryAmount, i.e.

```java
javax.money.Monetary.getDefaultAmountFactory()
                    .setNumber(number)
                    .setCurrency(currency)
                    .create()
```

So something like
```suggestion
        deserializers.addDeserializer(MonetaryAmount.class, new MonetaryAmountDeserializer<>((amount, currency) -> Monetary.getDefaultAmountFactory()
                                                             .setNumber(number)
                                                             .setCurrency(currency)
                                                             .create(),
                                 names));
```

Reply:
Thanks for the pointer. I edited the method to use the SPI capabilities.

Reply:
I didn't see the `amountFactory` field at first but I now get its usage.
Maybe the constructor passing a null `amountFactory` should get a lambda based on `Monetary.getDefaultAmountFactory()...create()` instead ? This way you would unify the logic (no need to check if `amountFactory` is null or not

From this 
```java
public JavaxMoneyModule() {
        this(new DecimalAmountWriter(), FieldNames.defaults(), MonetaryAmountFormatFactory.NONE, null, null);
}
```
to this
```java
public JavaxMoneyModule() {
        this(new DecimalAmountWriter(), FieldNames.defaults(), MonetaryAmountFormatFactory.NONE, null, (amount, currency) -> Monetary.getAmountFactory(c)
                                                                                                                                     .setNumber(amount)
                                                                                                                                     .setCurrency(currency)
                                                                                                                                     .create());
}
```

Reply:
Thanks for the suggestion. I have refactored as mentioned.